crates/ruma-state-res/src/event_auth.rs

```diff
@@ -343,10 +345,19 @@ pub fn auth_check<E: Event>(
         }
     } else {
         // If no power level event found the creator gets 100 everyone else gets 0
-        from_json_str::<RoomCreateEventContent>(room_create_event.content().get())
-            .ok()
-            .and_then(|create| (create.creator == *sender).then(|| int!(100)))
-            .unwrap_or_default()
+        let is_creator = if room_version.use_room_create_sender {
+            room_create_event.sender() == sender
+        } else {
+            #[allow(deprecated)]
+            from_json_str::<RoomCreateEventContent>(room_create_event.content().get())
```

Comment:
I'm wondering if we could use [`SingleFieldVisitor`](https://github.com/ruma/ruma/blob/ffdd6327015c852427b927084f038e89f001e896/crates/ruma-common/src/serde/raw.rs#L139) here, since we don't need the whole event. Especially if we could just get the borrowed type.

Reply:
Seems worth a separate PR / discussion on Matrix. It's not clear to me what you mean by "the borrowed type".